lib/liquid.lua

```diff
@@ -2547,28 +2549,39 @@ end
 -------------------------------------------------------------ParserContext end ---------------------------------------------------------
 -------------------------------------------------------------FileSystem begin ---------------------------------------------------------
 -- local FileSystem = {}
-function FileSystem:new( location )
+function FileSystem:new( get, error_handler )
     -- body
     local instance = {}
     setmetatable(instance, {__index = FileSystem})
-    instance.location = location
+    instance.get = get
     instance.text = nil
+    instance.error_handler = error_handler
     return instance
 end
 --
-function FileSystem:genertic_get( ... )
+function FileSystem:genertic_get( location )
+    local error_handler = self.error_handler
     -- body
     if self.get and type(self.get) == "function" then
-        local status, err = pcall(self.get, self.location)
-        if status then
-            return err
+        local ok, ret = pcall(self.get, location)
+
+        if not ret then
```

Comment:
Indeed. I expected `tostring({})` will throw and error because that is what `ngx.log` does when you give it table without `__tostring` metamethod. https://github.com/openresty/lua-nginx-module/blob/a5094766702d97ab43c742f26d6d55098aa1b1f0/src/ngx_http_lua_log.c#L165-L168

Ok. Probably good idea to not covert to string then. 

Allowing passing table with proper method could be done in the future when there is actually need for it.